Modules/Video/Core/include/itkImageToVideoFilter.hxx

```diff
@@ -0,0 +1,265 @@
+  // change the requested regions on the input so we need a non-const version
+  this->SetInput(0, const_cast<InputImageType *>(image));
+}
+
+//
+// SetInput( idx, videoStream )
+//
+template <typename TInputImage, typename TOutputVideoStream>
+void
+ImageToVideoFilter<TInputImage, TOutputVideoStream>::SetInput(unsigned int idx, const TInputImage * image)
+{
+  // We keep this const_cast because in actuality, we do want to be able to
+  // change the requested regions on the input so we need a non-const version
+  this->TemporalProcessObject::SetNthInput(idx, const_cast<InputImageType *>(image));
+}
+
+//
+// GetInput()
+//
+template <typename TInputImage, typename TOutputVideoStream>
+const TInputImage *
+ImageToVideoFilter<TInputImage, TOutputVideoStream>::GetInput() const
+{
+  if (this->GetNumberOfInputs() < 1)
+  {
+    return nullptr;
+  }
+  return static_cast<const InputImageType *>(this->ProcessObject::GetInput(0));
+}
+
+//
+// GetInput(idx)
+//
+template <typename TInputImage, typename TOutputVideoStream>
+const TInputImage *
+ImageToVideoFilter<TInputImage, TOutputVideoStream>::GetInput(unsigned int idx) const
+{
+  return static_cast<const InputImageType *>(this->ProcessObject::GetInput(idx));
+}
+
+//-PROTECTED METHODS-----------------------------------------------------------
+
+//
+// GetInput (non-const)
+//
+template <typename TInputImage, typename TOutputVideoStream>
+TInputImage *
+ImageToVideoFilter<TInputImage, TOutputVideoStream>::GetInput()
+{
+  return GetInput(0);
+}
+
+//
+// GetInput (non-const)
+//
+template <typename TInputImage, typename TOutputVideoStream>
+TInputImage *
+ImageToVideoFilter<TInputImage, TOutputVideoStream>::GetInput(unsigned int idx)
+{
+  return static_cast<InputImageType *>(this->ProcessObject::GetInput(idx));
+}
+
+//
+// GenerateOutputInformation
+//
+template <typename TInputImage, typename TOutputVideoStream>
+void
+ImageToVideoFilter<TInputImage, TOutputVideoStream>::GenerateOutputInformation()
+{
+  // Get the input
+  const InputImageType * input = this->GetInput();
+
+  // Get first input frame's largest possible spatial region
+  InputImageRegionType inputRegion = input->GetLargestPossibleRegion();
+
+  // Set temporal frame + duration from user-defined frame axis in input image
+  OutputTemporalRegionType outputTemporalRegion;
+  outputTemporalRegion.SetFrameStart(inputRegion.GetIndex(m_FrameAxis));
+  outputTemporalRegion.SetFrameDuration(inputRegion.GetSize(m_FrameAxis));
+  // TODO set real start and real duration
+  this->GetOutput()->SetLargestPossibleTemporalRegion(outputTemporalRegion);
+
+  // Set the output spatial region from the input image's largest spatial region,
+  // discarding along the user-defined image axis
+  OutputFrameSpatialRegionType                          outputSpatialRegion;
+  typename OutputFrameSpatialRegionType::IndexValueType outputIdx = 0;
+  for (typename itk::IndexValueType idx = 0; idx < InputImageRegionType::ImageDimension; idx++)
```

Comment:
I am surprised `typename` is needed here.

Reply:
It was not 😃